File: programmers/외계어 사전.py
def solution(spell, dic):
    answer = 0

    for d in dic:
        cnt = ''
        for s in spell:
            if d.count(s) == 1:
                cnt += '1'
        if len(cnt) == len(spell):
            answer += 1
    return 2 if answer==0 else answer


print(solution(["p", "o", "s"], ["sod", "eocd", "qixm", "adio", "soo"]))
print(solution(["z", "d", "x"], ["def", "dww", "dzx", "loveaw"]))
print(solution(["s", "o", "m", "d"], ["moos", "dzx", "smm", "sunmmo", "som"]))

File: programmers/코딩테스트 입문 with Java/외계어사전.java
package 코딩테스트입문;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120869
 */
public class 외계어사전 {
    public static void main(String[] args) {
        System.out.println(solution(new String[]{"p", "o", "s"}, new String[]{"sod", "eocd", "qixm", "adio", "soo"}));
        System.out.println(solution(new String[]{"z", "d", "x"}, new String[]{"def", "dww", "dzx", "loveaw"}));
        System.out.println(solution(new String[]{"s", "o", "m", "d"}, new String[]{"moos", "dzx", "smm", "sunmmo", "som"}));
    }

    public static int solution(String[] spell, String[] dic) {
        int answer = 0;
        List<String> list = new ArrayList<>();
        String[] str;
        Arrays.sort(spell);
        for (String s : dic) {
            if (s.length() == spell.length) {
                list.add(s);
            }
            str = s.split("");
            Arrays.sort(str);
            if (Arrays.equals(str, spell)) {
                answer++;
            }
        }
        if (answer == 0) {
            answer = 2;
        }
        return answer;
    }
}
